src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let amount = 0;
+
+  const adder = (a) => {
+    if (a !== undefined) {
+      amount += a;
+
+      return adder;
+    } else {
+      const result = amount;
```

Comment:
Copying the `amount` value into a separate `result` variable is unnecessary here. You can directly return `amount` and then reset it to `0` on the next line.